File: ZP_Base/funcCoding/funcCoding/ViewController.m
//
//  ViewController.m
//  funcCoding
//
//  Created by zhangpoor on 16/11/15.
//  Copyright © 2016年 zhangpoor. All rights reserved.
//

#import "ViewController.h"


typedef id (^FuncA_callback)(int p1);




@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    FuncA_callback fa_001 = ^id(int p1){
      
        //NSString *_str = p1;
        
        //return _str;
        
        return ^NSString *(NSString *p2)
        {
            NSString *_str = [p2 stringByAppendingFormat:@":%d",p1];
            
            return _str;
        };
    };
    
    
    NSString *sss = fa_001(1)(@"abc");
    
    NSLog(@"%@",sss);
    NSLog(@"aaaaa");
}




@end

File: ZP_Base/SF001/SF001/ViewController.swift
//
//  ViewController.swift
//  SF001
//
//  Created by zhangpoor on 16/10/20.
//  Copyright © 2016年 zhangpoor. All rights reserved.
//

import UIKit

import F001

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        print("bbbb");
        
        let c:ClassA =  ClassA();
        
        let s:String = c.f001(keyPath: "abc");
        
        
        print("\(s)");
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

